### PR TITLE
Create deb package on pull requests

### PR DESCRIPTION
**Pull request by ronakfof**
*Issue #, if available:*

*Description of changes:*
With this change, now a deb package can be created and upload to artifacts for Pull Requests labeled with `create-deb-package`

[Example](https://github.com/awslabs/aws-build-accumulator/actions/runs/2096133094)

*How to fetch artifact?*
```
Download manually and use
```
OR

Crete a file called `_netrc` with following syntax - 
```
default login <USERNAME> password <PERSONAL ACCESS TOKEN>
```
and then -
```
curl --netrc-file PATH_TO_NETRC \
       -o artifact_name.zip \
       -L https://api.github.com/repos/awslabs/aws-build-accumulator/actions/artifacts/ARTIFACT_ID/zip
```
which can then be unzipped to get the `deb package`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by nrdg42** [comment]
What's the use case of this feature?

**Comment by angelonakos** [comment]
> What's the use case of this feature?

This "create-deb-package" label enables downstream customers of Litani to create experimental packages and to test out these types of packages with regards to their use case without our team having to merge a change or cut a release.

**Review by angelonakos — APPROVED** [review]

File: .github/actions/build/action.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        sudo apt-get install -y mandoc scdoc ninja-build
+        mkdir -p litani-${{ inputs.version }}/{DEBIAN,usr/{bin,libexec/litani,share/{doc/litani,man/{man1,man5,man7}}}}
+        touch litani-${{ inputs.version }}/DEBIAN/control
+        cat << EOF > litani-${{ inputs.version }}/DEBIAN/control
+        Package: Litani
+        Version: ${{ inputs.version }}
+        Architecture: amd64
+        Depends: ninja-build, gnuplot, graphviz, python3-jinja2
+        Maintainer: Kareem Khazem <karkhaz@amazon.co.uk>
+        Description: A program that provides platform-independent job control.
```
Comment (karkhaz) [inline .github/actions/build/action.yaml]: Could you use the latest text from the README?

**Review by karkhaz — CHANGES_REQUESTED** [review]
Nice, thank you, it's good to see it split into several files also. Does anything depend on the `release-packages.yaml` that was renamed to `actions/build/action.yaml`?

**Comment by ronakfof** [comment]
> Nice, thank you, it's good to see it split into several files also. Does anything depend on the `release-packages.yaml` that was renamed to `actions/build/action.yaml`?

`release-packages.yaml` is deleted. I separated both `release brew` and `release deb` process into 2 separate workflows. And since creation of deb package was being done in 2 workflows (release and on pr), I abstracted it into `actions/build/action.yaml`. So both `release-deb.yaml` and `create-deb-on-pr.yaml` depend on it now

File: .github/actions/build/action.yaml (hunk 1 above)
Comment (ronakfof) [reply .github/actions/build/action.yaml]: Resolved

**Review by karkhaz — APPROVED** [review]

**Pull request merged**